modules/azure/iam_set/main.tf

```diff
@@ -0,0 +1,37 @@
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.48"
+    }
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+locals {
+  role_assignments = flatten([
+    for principal_key, principal_id in var.principals :
+    [
+      for role in var.roles : {
+        scope                = role.scope
+        role_definition_name = role.role_name
+        principal_id         = principal_id
+      }
+    ]
+  ])
+}
+
+resource "azurerm_role_assignment" "role_assignment" {
+  for_each = { for idx, ra in local.role_assignments : idx => ra }
```

Comment:
Can you tell me what the value of idx is?
Because if this is just the index number of the list i'm not going to allow it.

Reply:
You are right, I've corrected and removed the index number